capplets/appearance/appearance-font.c

```diff
@@ -49,61 +43,46 @@
 #define DPI_FALLBACK 96
 #define DPI_LOW_REASONABLE_VALUE 50
 #define DPI_HIGH_REASONABLE_VALUE 500
-#endif /* HAVE_XFT2 */
 
 static gboolean in_change = FALSE;
 
-#define MAX_FONT_POINT_WITHOUT_WARNING 32
-#define MAX_FONT_SIZE_WITHOUT_WARNING MAX_FONT_POINT_WITHOUT_WARNING * 1024
-
-#ifdef HAVE_XFT2
-
-#if !GTK_CHECK_VERSION (3, 0, 0)
-/*
- * Code for displaying previews of font rendering with various Xft options
- */
-
-static void sample_size_request(GtkWidget* darea, GtkRequisition* requisition)
-{
-	GdkPixbuf* pixbuf = g_object_get_data(G_OBJECT(darea), "sample-pixbuf");
-
-	requisition->width = gdk_pixbuf_get_width(pixbuf) + 2;
-	requisition->height = gdk_pixbuf_get_height(pixbuf) + 2;
-}
-#endif
-
 #if GTK_CHECK_VERSION (3, 0, 0)
 static void sample_draw(GtkWidget* darea, cairo_t* cr)
 #else
 static void sample_expose(GtkWidget* darea, GdkEventExpose* expose)
 #endif
 {
-	GtkAllocation allocation;
-	GdkPixbuf* pixbuf = g_object_get_data(G_OBJECT(darea), "sample-pixbuf");
-	int width = gdk_pixbuf_get_width(pixbuf);
-	int height = gdk_pixbuf_get_height(pixbuf);
+	cairo_surface_t* surface = g_object_get_data(G_OBJECT(darea), "sample-surface");
 
-	gtk_widget_get_allocation (darea, &allocation);
-	int x = (allocation.width - width) / 2;
-	int y = (allocation.height - height) / 2;
+	int x, y, w, h;
 
-	GdkColor black, white;
-	gdk_color_parse ("black", &black);
-	gdk_color_parse ("white", &white);
+#if GTK_CHECK_VERSION (3, 0, 0)
+	x = gtk_widget_get_allocated_width (darea);
+	y = gtk_widget_get_allocated_height (darea);
+#else
+	GtkAllocation allocation;
+	gtk_widget_get_allocation (darea, &allocation);
+	x = allocation.width;
+	y = allocation.height;
+#endif
+	w = cairo_image_surface_get_width (surface);
+	h = cairo_image_surface_get_height (surface);
```

Comment:
Afaics gtk_widget_get_allocated_height/width are just convenience functions. So let's get rid of the if/else here and just use a GtkAllocation to query width and height across Gtk versions.